Sources/RepoToolsCore/RepoActions.swift

```diff
@@ -399,15 +401,34 @@ public enum RepoActions {
         let extractDir = shell.tmpdir()
         func extract() -> CommandOutput {
             let lowercasedFileName = fileName.lowercased()
-            if lowercasedFileName.hasSuffix("zip") {
+            if fetchOptions.fileHandler?.isEmpty == false {
```

Comment:
I am not a swift expert, so there may be a more elegant way of doing this cascading logic...  But I wanted to treat any definition of this override as different with no falling back (that way if the developer does define an invalid override it doesn't quietly fail and carry forward to extension checking).

Reply:
I would do something like this:

```swift
let resolvedExtractionHandler: String
if fetchOptions.fileHandler == "default" {
    let lowercasedFileName = fileName.lowercased()
    if lowercasedFileName.hasSuffix("zip") {
        resolvedExtractionHandler = "zip"
    } else if lowercasedFileName.hasSuffix("tar.gz") || lowercasedFileName.hasSuffix("tgz") {
        resolvedExtractionHandler = "tar"
    } else {
        fatalError("Cannot extract files other than .zip, .tar.gz, or .tgz")
    }
} else {
    resolvedExtractionHandler = fetchOptions.fileHandler
}

if resolvedExtractionHandler == "zip" {
    return shell.command(CommandBinary.sh, arguments: [
        "-c",
        unzipTransaction(
            rootDir: extractDir,
            fileName: escape(download)
        ),
    ])
} else if resolvedExtractionHandler == "tar" {
    return shell.command(CommandBinary.sh, arguments: [
        "-c",
        untarTransaction(
            rootDir: extractDir,
            fileName: escape(download)
        ),
    ])
}
fatalError("Should not be possible to get here.")
```

Reply:
So I broke the logic down about the same way this is done: https://github.com/discordapp/PodToBUILD/pull/1/commits/43c63479fa365a25e868078b4768303f85124ded#diff-8899a6f8f09176941c964bb6deee3f2cR404